src/bin/ion/commands/generate/templates/java/scalar.templ

```diff
@@ -1,69 +1,69 @@
     }
 
     /**
-     * Reads a {{ target_kind_name }} from an {@link IonReader}.
+     * Reads a {{ model.name }} from an {@link IonReader}.
      *
      * This method does not advance the reader at the current level.
      * The caller is responsible for positioning the reader on the value to read.
      */
-    public static {{ target_kind_name }} readFrom(IonReader reader) {
+    public static {{ model.name }} readFrom(IonReader reader) {
         {# Initializes all the fields of this class #}
-        {{ fields[0].value_type }} value =
-        {% if fields[0].value_type == "boolean" %}
+        {{ base_type }} value =
+        {% if base_type == "boolean" %}
             false
-        {% elif fields[0].value_type == "int" or fields[0].value_type == "double" %}
+        {% elif base_type == "int" or base_type == "double" %}
             0
         {% else %}
             null
         {% endif %};
         {# Reads `Value` class with a single field `value` #}
-        value = {% if fields[0].value_type | is_built_in_type %}
-                    {% if fields[0].value_type == "bytes[]" %}
+        value = {% if base_type | is_built_in_type %}
+                    {% if base_type == "bytes[]" %}
                         reader.newBytes();
                     {% else %}
-                        reader.{{ fields[0].value_type | camel }}Value();
+                        reader.{{ base_type | camel }}Value();
                     {% endif %}
                  {% else %}
-                    {{ fields[0].value_type }}.readFrom(reader);
+                    {{ base_type }}.readFrom(reader);
                  {% endif %}
-        {{ target_kind_name }} {{ target_kind_name | camel }} = new {{ target_kind_name }}();
-        {{ target_kind_name | camel }}.value = value;
+        {{ model.name }} {{ model.name | camel }} = new {{ model.name }}();
+        {{ model.name | camel }}.value = value;
 
-        return  {{ target_kind_name | camel }};
+        return  {{ model.name | camel }};
     }
 
     /**
-     * Writes a {{ target_kind_name }} as Ion from an {@link IonWriter}.
+     * Writes a {{ model.name }} as Ion from an {@link IonWriter}.
      *
      * This method does not close the writer after writing is complete.
      * The caller is responsible for closing the stream associated with the writer.
      */
     public void writeTo(IonWriter writer) throws IOException {
         {# Writes `Value` class with a single field `value` as an Ion value #}
-        {% if fields[0].value_type | is_built_in_type == false  %}
+        {% if base_type | is_built_in_type == false  %}
             this.value.writeTo(writer)?;
         {% else %}
-            writer.write{{ fields[0].isl_type_name | upper_camel }}(this.value);
+            writer.write{{ base_type | replace(from="double", to="float") | replace(from="boolean", to="bool") | upper_camel }}(this.value);
         {% endif %}
     }
 }
+{% endmacro %}
+{{ self::scalar(model=model) }}
```

Comment:
I am using this same template to render the nested type instead of creating another template for nested types. Which is why I had to convert these templates to macro, so they can also be called whenever a nested type is found.